python/nx-cugraph/nx_cugraph/classes/graph.py

```diff
@@ -458,6 +458,24 @@ def has_edge(self, u: NodeKey, v: NodeKey) -> bool:
                 return False
         return bool(((self.src_indices == u) & (self.dst_indices == v)).any())
 
+    def _neighbors(self, n: NodeKey) -> cp.ndarray[NodeValue]:
+        if n not in self:
+            hash(n)  # To raise TypeError if appropriate
+            raise nx.NetworkXError(
+                f"The node {n} is not in the {self.__class__.__name__.lower()}."
+            )
+        if self.key_to_id is not None:
+            n = self.key_to_id[n]
+        nbrs = self.dst_indices[self.src_indices == n]
+        if self.is_multigraph():
+            nbrs = cp.unique(nbrs)
+        return nbrs
+
+    @networkx_api
+    def neighbors(self, n: NodeKey) -> Iterator[NodeKey]:
+        nbrs = self._neighbors(n)
```

Comment:
I'm curious about this pattern here; why is there a separate `_neighbors` method instead of just including it inline here in `neighbors`?

Reply:
Different return type. Sometimes we want the cupy array. `G.neighbors` is part of the networkx API and is an iterator.